tests/ui/cfg/diagnostics-reexport.rs

```diff
@@ -1,7 +1,7 @@
 pub mod inner {
-    #[cfg(FALSE)]
+    #[cfg(FALSE)] //~ NOTE the item is gated here
```

Comment:
I don't think this is correct? no one ever references this item.

Reply:
This is because `uwu` under `inner::gone` now gets included for `inner` (because `gone` is `cfg`d out) when trying to access `inner::uwu`. I believe this is the right thing to do. If the user had written `inner::gone::uwu`, the output is:

```
error[E0433]: failed to resolve: could not find `gone` in `inner`
  --> tests/ui/cfg/diagnostics-reexport.rs:38:12
   |
38 |     inner::gone::uwu(); //~ ERROR cannot find function
   |            ^^^^ could not find `gone` in `inner`
   |
note: found an item that was configured out
  --> tests/ui/cfg/diagnostics-reexport.rs:3:9
   |
3  |     mod gone {
   |         ^^^^
note: the item is gated here
  --> tests/ui/cfg/diagnostics-reexport.rs:2:5
   |
2  |     #[cfg(FALSE)] //~ NOTE the item is gated here
   |     ^^^^^^^^^^^^^
```

but if it is `inner::gon::uwu` then the output is

```
error[E0433]: failed to resolve: could not find `gon` in `inner`
  --> tests/ui/cfg/diagnostics-reexport.rs:38:12
   |
38 |     inner::gon::uwu(); //~ ERROR cannot find function
   |            ^^^ could not find `gon` in `inner`
```

We might actually want to still search for `cfg`d out items both with levenshtein distance (but the signal there is much weaker) and by looking at the very last path segment (to find the `cfg`d out `uwu`).

Reply:
I tend to agree with @Noratrieb: the user wrote `inner::uwu`, which refers to the `use super::uwu` below, which already failed to resolve without suggesting `gone`. Either both suggest `gone::uwu`, or none of them.